linode/manager
# refactor: [M3-8291] - Enforce Factory.each to start increment at `1`

## Description 📝
This PR makes sure `Factory.each` always start iterating from `1` rather than `0`. Having falsy entity IDs is responsible for for a few issues (unit tests false positives in, disabling queries based on a truthy entity id) and will help us with our "Next Gen MSW ©" where we will generate new entities based on factories and those can't have `0` as an id.

In order to do so I added a monkey patch (run time modifier) for `Factory.each` so that it increments from `1` in every factory without having to modify every factory instance. This means not only IDs but also every value relying on an `each` was modified. As a result, many tests asserting either for some order, value or array length needed to be updated. It also surfaced a few issues with some tests, tho more could have been fixed (outside of scope for this PR).

This PR does not touch any consumer code.

## Changes  🔄
- Patch Factory.js `each` to increment from 1
- Fix tests affected

## How to test 🧪

### Verification steps
- run `yarn test` and confirm all tests are passing locally and in CI
- confirm e2e suites are passing in CI

## As an Author I have considered 🤔

*Check all that apply*
